### PR TITLE
Forcer l'update du mot de passe des utilisateurs -> vers feature/professionnals-password-length

### PR DESCRIPTION
**Pull request by AntoineAresu**
[Ticket](https://trello.com/c/Blp8gIBt/4681-5-forcer-lupdate-du-mot-de-passe-des-utilisateurs)

### CODE REVIEW

File: src/EventSubscriber/AbstractGDPRSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+use App\Entity\User;
+use App\Security\HelperV2\Oauth2Helper;
+use App\ServiceV2\Traits\UserAwareTrait;
+use League\Bundle\OAuth2ServerBundle\Security\Authentication\Token\OAuth2Token;
+use Symfony\Bundle\SecurityBundle\Security;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+abstract class AbstractGDPRSubscriber implements EventSubscriberInterface
```
Comment (NilsRec) [inline src/EventSubscriber/AbstractGDPRSubscriber.php:14]: I'm not sure that the name is right. I understand that we use it in GDPR issues context, but when you look at the class itself, it does not have anything to do with GDPR

File: src/EventSubscriber/PasswordPolicySubscriber.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+
+class PasswordPolicySubscriber extends AbstractGDPRSubscriber implements EventSubscriberInterface
+{
+    private const ALLOWED_ROUTES = ['improve_password'];
+
+    public function checkUserHasUpdatedWithStrongPassword(RequestEvent $event): void
```
Comment (NilsRec) [inline src/EventSubscriber/PasswordPolicySubscriber.php:13]: `checkUserHasUpdatedWithLatestPAsswordPolicy` ?

File: src/Security/Authenticator.php [hunk 3]
```diff
@@ -50,6 +52,9 @@ public function onAuthenticationSuccess(Request $request, TokenInterface $token,
         $previousLogin = $user->getDerniereConnexionAt()?->format('Y-m-d');
         $now = new \DateTime();
         $user->setLastIp($request->getClientIp())->setDerniereConnexionAt($now);
+        if ($this->passwordHelper->isStrong($request->request->get('_password'), $user->isBeneficiaire())) {
```
Comment (NilsRec) [inline src/Security/Authenticator.php:55]: `if (!$user->hasPasswordWithLatestPolicy() && $this->passwordHelper->isStrong($request->request->get('_password'), $user->isBeneficiaire()))`
So we don't have to check password strength each time

File: tests/v2/Controller/GdprController/ImprovePasswordTest.php [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        bool $isXmlHttpRequest = false,
+        array $body = [],
+    ): void {
+        $this->assertRoute($url, $expectedStatusCode, $userMail, $expectedRedirect, $method);
+    }
+
+    public function provideTestRoute(): \Generator
+    {
+        yield 'Should redirect to login when not authenticated' => [self::URL, 302, null, '/login'];
+        yield 'Should return 200 status code when authenticated as member' => [self::URL, 200, MemberFixture::MEMBER_MAIL];
```
Comment (NilsRec) [inline tests/v2/Controller/GdprController/ImprovePasswordTest.php:32]: ~~member~~ --> pro

**Review by tcheymol — APPROVED** [review]

**Pull request merged**